### ISSUE TITLE
Behavior of scope=page-content/start-offset=-1 on first page of volume/document

### ISSUE THREAD

**Issue by bertfrees** [joeha480/dotify#144]
Disclaimer: I'm not saying the behavior is wrong, I'm using this issue to explain my problem.

The way I have implemented the CSS keyword [`spread-start`](http://snaekobbi.github.io/braille-css-spec/master/index.html#dfn-string-spread-start) currently is like this (a bit inspired by the way Joel does it in the MTM script):
- Everywhere I put a `<marker name="foo" value="x"/>` I also put a `<marker name="foo/prev"/>` with the value of the previous marker named "foo".
- In order to find the `spread-start` value of "foo" I first search forward on the `page-content` (with an offset of -1 when we're on a right page) for "foo/prev" markers,
- and if nothing is found I search backward in the `sequence` for "foo" markers (again with an offset of -1 when on a right page).

Firstly, the backward search should be in scope document/volume, as soon as that is supported.  

Secondly, this method fails on the first page of the document or volume, i.e. on a right page when there is no left page (and currently also on the first page on a sequence, see issue #143). The reason is that when a value is assigned at the start of the document, the current implementation will behave as if it was assigned in the middle of the spread, and therefore no marker is found.

The way I currently see it, there are 3 solutions:
- For the backward search, Use scope `sequence` but without a `start-offset`. However this means I would have to use something (unexisting) called `spread-content` for the forward search.
- This implementation of `spread-start` is a bit hacky, so we could also consider supporting it more directly in OBFL/Dotify.
- I can create special page templates for the first pages of volumes. However for this to work selecting the first page in a volume must be supported in the `use-when` expression.

**Comment by bertfrees** [comment]
I will add a test for this if you want.

**Comment by joeha480** [comment]
@bertfrees Maybe it will be clearer with at test

**Comment by joeha480** [comment]
Do you want to return values from page 1 on both page 1 and 2 but otherwise n-1 (where n is the page number), is that it?

**Comment by bertfrees** [comment]
> Do you want to return values from page 1 on both page 1 and 2 but otherwise n-1 (where n is the page number), is that it?

Yes, that is it, and it does indeed make little sense to assume that. I've written a little test that confirms the behavior is what it is. See [marker-reference-start-offset-first-page](https://rawgit.com/snaekobbi/dotify/168c50fe265c9b007cf5557b95b0b1bf4fe44201/dotify.formatter.impl/test/org.daisy.dotify.engine.impl.resource-files.tests-overview.html#dp2/marker-reference-start-offset-first-page-input.obfl).

However I'm still here with my problem of implementing `spread-start` correctly (the reason I created this issue). I have made a test that should make more clear what I want to accomplish. See [marker-reference-spread-start-workaround](https://rawgit.com/snaekobbi/dotify/168c50fe265c9b007cf5557b95b0b1bf4fe44201/dotify.formatter.impl/test/org.daisy.dotify.engine.impl.resource-files.tests-overview.html#dp2/marker-reference-spread-start-workaround-input.obfl) (note that the OBFL and PEF files have some comments in it that should help you understand better what is going on in the test).

The 3 solutions I can come up with are still:
- add scope `spread-content`
- more direct support for CSS value `spread-start`
- create special template for first page, if the `use-when` expression supports this

I will have a look at this myself, but which one do you think makes most sense?

**Comment by joeha480** [comment]
I'll look into the details when I get a chance, but without looking into the details I can say I think a special template for the first page should definitely work, just make sure you put it first in the document (as it picks the first one that matches).
`use-when="(= $page 1)"`

**Comment by bertfrees** [comment]
Thanks. But are you sure the use-when expression will work? What is the exact definition of $page? Will it be 1 on the first page of every volume? Is the value independent of initial-page-number (see https://github.com/joeha480/dotify/issues/134)?

**Comment by joeha480** [comment]
No, in that case, you're right. It is not what you want. But maybe it is still easier to add another variable?

Related to:
https://github.com/braillespecs/obfl/issues/20

**Comment by bertfrees** [comment]
OK adding another variable makes sense. I'll try that.

**Comment by bertfrees** [comment]
Adding a variable "page-in-volume" turns out to be problematic because it requires the page breaking and volume breaking to happen at the same time.

**Comment by bertfrees** [comment]
I ended up implementing this with a forward/spread search and a backward/sequence search: https://github.com/daisy/pipeline-mod-braille/commit/b389baebe6da11889e4d76d52363344b36b669ee.
